src/components/UserSelector.tsx

```diff
@@ -1,62 +1,72 @@
-import React, { useContext, useEffect, useState } from 'react';
-import classNames from 'classnames';
-import { UserContext } from './UsersContext';
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as usersActions from '../features/users';
+import * as postsActions from '../features/posts';
 import { User } from '../types/User';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
+export const UserSelector: React.FC = memo(function UserSelectorComponent() {
+  const dispatch = useAppDispatch();
+
+  const { users, selectedUser, selectedPost } = useAppSelector(state => ({
+    ...state.users,
+    ...state.posts,
+  }));
+
+  const selectUser = useCallback(
+    (newUser: User) => () => {
+      if (selectedUser?.id === newUser.id) {
+        return;
+      }
+
+      if (selectedPost) {
```

Comment:
same as my comment in PostList - do not fetch posts here in this function and this component (it should work with user only). You can just keep the original structure and fetch posts for user in useEffect in App.tsx